lambda/matching.ml

```diff
@@ -680,7 +680,7 @@ end = struct
               | exception OrPat -> (
                   match p.pat_desc with
                   | Tpat_or (p1, p2, _) ->
-                      filter_rec [ p1 :: ps; p2 :: ps ] @ rem
+                      filter_rec ((p1 :: ps) :: (p2 :: ps) :: rem)
```

Comment:
@Octachron signals that this is not quite right with the current version of the function, which still has `let rem = filter_rec rem`. I will rewrite this commit to get rid of the `let rem = ...` which is un-necessarily confusing.

Reply:
Done.